tdigest/include/tdigest.hpp

```diff
@@ -297,6 +310,28 @@ class tdigest {
   static inline void check_split_points(const T* values, uint32_t size);
 };
 
+template<typename T, typename A>
+class tdigest<T, A>::const_iterator {
+public:
+  using iterator_category = std::input_iterator_tag;
+  using value_type = std::pair<const T&, const W>;
+  using difference_type = void;
+  using pointer = const return_value_holder<value_type>;
+  using reference = const value_type;
+
+  const_iterator(const tdigest<T, A> &tdigest_, bool is_end);
```

Comment:
I think this constructor should be private

Reply:
and <T, A> seems unnecessary again.
and our style is to attach & to the type, not to the name.
should be the last nit to pick :)

Reply:
Done